irc/handlers.go

```diff
@@ -2193,10 +2193,11 @@ func userHandler(server *Server, client *Client, msg ircmsg.IrcMessage, rb *Resp
 		return false
 	}
 
-	err := client.SetNames(msg.Params[0], msg.Params[3])
+	ident := msg.Params[0][:server.Limits().IdentLen-1] // -1 as SetNames adds the ~ at the start for us
```

Comment:
slice issue

(confirmed a panic in dev with a line like `USER a 0 * a`

Reply:
should be resolved now, added an additional check here. if there's a cleaner way to do this though, all the better